content/components/button-group.mdx

```diff
@@ -0,0 +1,100 @@
+---
+title: Button group
+description: Button group renders a series of buttons.
+reactId: button_group
+figma: https://www.figma.com/file/GCvY3Qv8czRgZgvl1dG6lp/Primer-Web?node-id=20363%3A71038&t=GrA68FvIJRoUaVbl-1
+rails: https://primer.style/view-components/components/beta/buttongroup
+---
+
+import {Box, Text} from '@primer/react'
+import ComponentLayout from '~/src/layouts/component-layout'
+export default ComponentLayout
+
+<img
+    src="https://user-images.githubusercontent.com/586552/226987730-84d99d46-1dd4-4ddc-91b3-046d7eeedbad.png"
+    role="presentation"
+    width="960"
+    alt="An image showing a button group in medium and small sizes."
+/>
+
+## Usage
+
+Buttons can be grouped together as individual segments of related actions. 
+
+Each segment in a button group is comprised from our default button component and can be visually represented with the same button types and states.
+
+Grouping buttons with a button group is better than rendering buttons close together for the following uses:
```

Comment:
Should we add something to explicitly mention grouping a button with a related dropdown? Or do you think these points already cover that?

For example:
![Screenshot 2023-03-23 at 2 27 24 PM](https://user-images.githubusercontent.com/2313998/227312220-7ec6f0df-6623-46ff-835d-6c49a701d8b3.png)

Reply:
This is probably enough: https://github.com/primer/design/pull/449/files#diff-364134a485fe7c28a5d968766c9d8fc4382644a06e0351f7809507e58d4a4289R63

We probably don't have to explicitly state that use-case here.